packages/blockchain-wallet-v4-frontend/src/services/SagaService/index.js

```diff
@@ -39,6 +39,20 @@ export const promptForInput = function*({ title, secret, initial = '' }) {
   }
 }
 
+export const confirm = function*({ message }) {
+  yield put(actions.modals.showModal('Confirm', { message }))
+  let { response, canceled } = yield race({
+    response: take(actionTypes.wallet.SUBMIT_CONFIRMATION),
+    canceled: take(actionTypes.modals.CLOSE_MODAL)
```

Comment:
our `closeModal` method could cause troubles closing not a target model but the last in stack. False `take` can happen here